github/clone_repositories.js

```diff
@@ -148,22 +125,34 @@ async function cloneRepositories(token, destination) {
       );
     }
 
-    const statusBar = new cliProgress.SingleBar(
-      {},
-      cliProgress.Presets.shades_classic,
-    );
+    const threads = [];
+
+    const statusBar = getStatusBar(repositories.length);
 
-    statusBar.start(repositories.length, 0);
+    statusBar.start();
 
     for (let j = 0; j < repositories.length; j += 1) {
       const repository = repositories[j];
-      await cloneRepository(repository, destination);
-      statusBar.update(j + 1);
+
+      const thread = clone
+        .run({ repository, destination })
+        .then(({ success, error }) => {
+          statusBar.update();
+
+          if (!success) {
+            failures.push(error);
+          }
+        });
```

Comment:
Isn't this possible instead of using return codes?

```suggestion
      const thread = clone
        .run({ repository, destination })
        .catch((err) => {
          failures.push(`${repository.fullName}: ${error.message}`)
        }).finally(() => {
          statusBar.update();
        })
```

Reply:
forgot about `finally`, much cleaner indeed 👍